arch/arm/core/cortex_m/prep_c.c

```diff
@@ -24,6 +24,45 @@
 #include <arch/arm/cortex_m/cmsis.h>
 #include <string.h>
 
+extern uint32_t _interrupt_stack;
+
+static inline void switch_sp_to_psp(void)
+{
+	__set_CONTROL(__get_CONTROL() | CONTROL_SPSEL_Msk);
+	/* Synchronous barrier to make psp new SP effective immediately */
+	__ISB();
+}
+
+static inline void set_and_switch_to_psp(void)
```

Comment:
perhaps `set_and_switch_to_use_psp`?

Reply:
both switch & use would be redundant...

Reply:
Well, I was thinking of "switching from using MSP to using PSP"
So it made sense to me :)

Reply:
Great ! Thanks for your review comments, appreciate it.